pipelines/template-build/template-build.yaml

```diff
@@ -121,6 +125,28 @@ spec:
       workspaces:
         - name: source
           workspace: workspace
+    - name: build-image-index
```

Comment:
Not sure it belongs here. To me it's more related to the docker-build pipeline.

Reply:
`template-build` is used as a template to create many of the different pipelines which generate container images. I think it makes sense to be here, but should we rename the template to be more precise about how it is intended?

The pipelines that use `template-build` are:
* docker-build
* docker-build-rhtap
* fbc-builder
* java-builder
* nodejs-builder
* tekton-bundle-builder (this is the outlier)

Reply:
I think that the proper resolution here is likely to move multiple tasks out of the `template-build` and into the `docker-build` pipeline. I think that this is a larger change and it should happen in a new PR to not grow this PR any larger.